cmake/CMakeLists.txt

```
@@ -179,28 +188,29 @@ if(BUILD_MPI)
     include(MPI4WIN)
   else()
     find_package(MPI REQUIRED)
-    include_directories(${MPI_CXX_INCLUDE_PATH})
-    add_definitions(-DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX=1)
-    list(APPEND LAMMPS_LINK_LIBS ${MPI_CXX_LIBRARIES})
+    target_compile_definitions(lammps PRIVATE -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX=1)
+    target_link_libraries(lammps PUBLIC MPI::MPI_CXX)
     option(LAMMPS_LONGLONG_TO_LONG "Workaround if your system or MPI version does not recognize 'long long' data types" OFF)
     if(LAMMPS_LONGLONG_TO_LONG)
-      add_definitions(-DLAMMPS_LONGLONG_TO_LONG)
+      target_compile_definitions(lammps PRIVATE -DLAMMPS_LONGLONG_TO_LONG)
     endif()
   endif()
 else()
   enable_language(C)
   file(GLOB MPI_SOURCES ${LAMMPS_SOURCE_DIR}/STUBS/mpi.c)
   add_library(mpi_stubs STATIC ${MPI_SOURCES})
-  include_directories(${LAMMPS_SOURCE_DIR}/STUBS)
-  list(APPEND LAMMPS_LINK_LIBS mpi_stubs)
+  install(TARGETS mpi_stubs  EXPORT LAMMPS_Targets LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  target_include_directories(mpi_stubs PUBLIC ${LAMMPS_SOURCE_DIR}/STUBS)
```

If this public, it definitely has to be wrapped in the generator expression. Not sure what is needed for linking to an installed liblammps, though. Maybe this?
```suggestion
  target_include_directories(mpi_stubs PUBLIC $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS> $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR>)
```
Though this would still be missing the `mpi.h`.